WordPress/Classes/ViewRelated/Media/CachedAnimatedImageView.swift

```diff
@@ -88,6 +100,21 @@ public class CachedAnimatedImageView: UIImageView, GIFAnimatable {
                                                               failure: failure)
     }
 
+    @objc public func setAnimatedImage(_ animatedImageData: Data,
+                                       success: (() -> Void)? = nil) {
+        currentTask?.cancel()
+        let didVerifyDataSize = gifPlaybackStrategy.verifyDataSize(animatedImageData)
+        if didVerifyDataSize {
+            animate(data: animatedImageData, success: success)
+        } else {
```

Comment:
Almost all this process is repeated inside the internal success block (L:84). Maybe we can extract it to share?